NadekoBot.Core/Modules/Utility/QuoteSlashCommands.cs

```diff
@@ -0,0 +1,380 @@
+using NadekoBot.Extensions;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System;
+using System.Net.Http;
+using Serilog;
+
+namespace NadekoBot.Modules.Utility
+{
+    public partial class Utility
+    {
+
+        public class QuoteSlashCommands : NadekoSlashSubmodule
+        {
+            private readonly DbService _db;
+            private readonly IHttpClientFactory _httpFactory;
+
+            public QuoteSlashCommands(DbService db,IHttpClientFactory factory)
+            {
+                _db = db;
+                _httpFactory = factory;
+            }
+
+
+            [NadekoSlash]
+            [RequireContext(ContextType.Guild)]
+            public async Task ListQuotes([Summary("page","page number")][MinValue(1)]int page = 1,
+            [Summary("order","Sorting Column")] OrderType order = OrderType.Keyword)
+            {
+                page -= 1;
+                if (page < 0)
+                    return;
+                int quotecount;
+                await ctx.Interaction.DeferAsync().ConfigureAwait(false);
+                using (var uow = _db.GetDbContext()){
+                    quotecount = uow.Quotes.GetGroupCount(ctx.Guild.Id);
+                }
+                await ctx.SendScrollingButtonAsync(page,(cur)=>{
+                IEnumerable<Quote> quotes;
+                using (var uow = _db.GetDbContext())
+                {
+                    quotes = uow.Quotes.GetGroup(ctx.Guild.Id, cur, order);
+                }
+
+                if (quotes.Any()){
+                    return new EmbedBuilder()
+                    .WithOkColor()
+                    .WithDescription(string.Join("\n", quotes.Select(q => $"`#{q.Id}` {Format.Bold(q.Keyword.SanitizeAllMentions()),-20} by {q.AuthorName.SanitizeAllMentions()}")))
+                    .WithTitle(GetText("quotes_page", cur + 1));
+                }else{
+                    return new EmbedBuilder()
+                    .WithErrorColor()
+                    .WithDescription(Format.Bold(ctx.User.ToString()) + " " + GetText("quotes_page_none"));
+                    
+                }},quotecount,15);
+
+            }
+
+            [NadekoSlash]
+            [RequireContext(ContextType.Guild)]
+            public async Task QuotePrint([Summary("keyword","Name of the quote")]string keyword)
+            {
+                if (string.IsNullOrWhiteSpace(keyword))
+                    return;
+
+                keyword = keyword.ToUpperInvariant();
+
+                Quote quote;
+                await ctx.Interaction.DeferAsync().ConfigureAwait(false);
+                using (var uow = _db.GetDbContext())
+                {
+                    quote = await uow.Quotes.GetRandomQuoteByKeywordAsync(ctx.Guild.Id, keyword);
+                    //if (quote != null)
+                    //{
+                    //    quote.UseCount += 1;
+                    //    uow.Complete();
+                    //}
```

Comment:
What's this commented code for?

Reply:
it was commented in QuoteCommands.cs and i just copied that. I can delete it if you want